default/python/AI/PlanetUtilsAI.py

```diff
@@ -180,10 +199,47 @@ def get_planet_opinion(feature: str) -> Opinion:
     return _calculate_get_planet_opinions().get(feature, default)
 
 
+@cache_for_current_turn
+def _planned_species() -> Mapping[PlanetId, str]:
+    universe = fo.getUniverse()
+    production_queue = fo.getEmpire().productionQueue
+    planned_species = {}
+    building_prefix = "BLD_COL_"
+    species_prefix = "SP_"
+    colonisation_plans = get_aistate().colonisablePlanetIDs
+    for element in production_queue:
+        if element.name.startswith(building_prefix):
```

Comment:
These string operations are mixed with prod inspection queue logic, Let's move it to a separate function.  _is_building(element).